### PR TITLE
[WIP] Traffic monitoring stats

### PR DESCRIPTION
**Pull request by zeptin**
This still needs improvement to be accurate for a long-running node, but it does at least give some insight into the level of traffic each peer is causing.

Also fixed some typos.

### CODE REVIEW

File: src/Stratis.Bitcoin/Connection/ConnectionManager.cs [hunk 1]
```diff
@@ -235,18 +237,24 @@ private void AddComponentStats(StringBuilder builder)
                 string peerHeights = $"(r/s):{(chainHeadersBehavior.BestReceivedTip != null ? chainHeadersBehavior.BestReceivedTip.Height.ToString() : peer.PeerVersion?.StartHeight + "*" ?? "-")}";
                 peerHeights += $"/{(chainHeadersBehavior.BestSentHeader != null ? chainHeadersBehavior.BestSentHeader.Height.ToString() : peer.PeerVersion?.StartHeight + "*" ?? "-")}";
 
+                // TODO: Need a snapshot cache so that not only currently connected peers are summed
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Connection/ConnectionManager.cs]: Aren't you already iterating over connection peers anyway?

**Comment by dangershony** [comment]
To get a correct measure we need to sample and keep historical data of peers that disconnected and reconnected overtime.
Perhaps we need to keep a sample table that goes back X days of historical data?

**Comment by zeptin** [comment]
@fassadlr
image

File: src/Stratis.Bitcoin/Connection/ConnectionManager.cs [hunk 2]
```diff
@@ -235,18 +237,24 @@ private void AddComponentStats(StringBuilder builder)
...
                     + peerHeights.PadRight(LoggingConfiguration.ColumnLength + 7)
+                    + peerTraffic.PadRight(LoggingConfiguration.ColumnLength + 7)
                     + " agent:" + agent);
             }
 
             int inbound = this.ConnectedPeers.Count(x => x.Inbound);
 
             builder.AppendLine();
-            builder.AppendLine($"======Connection====== agent {this.Parameters.UserAgent} [in:{inbound} out:{this.ConnectedPeers.Count() - inbound}]");
+            builder.AppendLine($"======Connection====== agent {this.Parameters.UserAgent} [in:{inbound} out:{this.ConnectedPeers.Count() - inbound}] [recv: {totalRead} sent: {totalWritten}]");
```
Comment (noescape00) [inline src/Stratis.Bitcoin/Connection/ConnectionManager.cs]: I'd also add avg traffic per minute
That will require keeping connection time of a peer (btw connection uptime is also smth we can display)

Comment (noescape00) [inline src/Stratis.Bitcoin/Connection/ConnectionManager.cs]: Maybe display in KB or MB?

Comment (dangershony) [reply src/Stratis.Bitcoin/Connection/ConnectionManager.cs]: yeha I prefer mb there is an override method that will convert it for you.

Comment (dangershony) [reply src/Stratis.Bitcoin/Connection/ConnectionManager.cs]: yeah good idea.

Comment (zeptin) [reply src/Stratis.Bitcoin/Connection/ConnectionManager.cs]: Done

**Pull request merged**